### ISSUE TITLE
Error with findOrCreate

### ISSUE THREAD

**Issue by pooleja** [cockroachdb/sequelize-cockroachdb#15]
I am trying to use sequelize findOrCreate() and it looks like it is throwing an exception only when the transaction is not found and it needs to created it.

Is this feature not supported?

Code
```
return Model.findOrCreate({
    where: data
  })
```

Error
```
Unhandled rejection SequelizeDatabaseError: syntax error at or near "or"
CREATE OR REPLACE FUNCTION pg_temp.testfunc(OUT response "data_log", OUT sequelize_caught_exception text) RETURNS RECORD AS $func_b1d03f43ce7e43ec921005a0ce931dee$ BEGIN INSERT INTO "data_log" ("data_1","data_2") VALUES ('string1','string2') RETURNING * INTO response; EXCEPTION WHEN unique_violation THEN GET STACKED DIAGNOSTICS sequelize_caught_exception = PG_EXCEPTION_DETAIL; END $func_b1d03f43ce7e43ec921005a0ce931dee$ LANGUAGE plpgsql; SELECT (testfunc.response).*, testfunc.sequelize_caught_exception FROM pg_temp.testfunc(); DROP FUNCTION IF EXISTS pg_temp.testfunc();
```

**Comment by JonasMS** [comment]
@benesch You close this issue and refer to #16 but then in #16 you also close the issue and refer to this issue (#15). . .

**Comment by benesch** [comment]
@JonasMS this issue isn't closed!

**Comment by JonasMS** [comment]
Oy vey, my bad. Seems that the "Closed" tag is referring to #16.

**Comment by benesch** [comment]
Heh, no problem at all. GitHub's UI is rather confusing here.

**Comment by internalfx** [comment]
This is also affecting [sqlgrid](https://github.com/internalfx/sqlgrid).

**Comment by zllovesuki** [comment]
At this moment, I'm using this as a workaround:
```javascript
let t = await sql.transaction()

try {
    // https://github.com/sequelize/sequelize/issues/8628
    try {
        chunk = await chunks.create({
            sha256: hash,
            data: data
        }, {
            transaction: t
        })
    } catch (e) {
        chunk = await chunks.findOne({
            where: {
                sha256: hash
            }
        }, {
            transaction: t
        })
    }
    await t.commit()
} catch (e) {
    await t.rollback()
}
```

**Comment by pashinin** [comment]
Can this be fixed so that we can use findOrCreate?

**Comment by brlocky** [comment]
> Can this be fixed so that we can use findOrCreate?

And still not fixed

**Comment by dustinsmith1024** [comment]
The `findCreateFind` method seems to be an option that works with CockRoach. Might not work in all cases.

https://sequelize.org/master/class/lib/model.js~Model.html#static-method-findCreateFind

**Comment by awoods187** [comment]
@rafiss can you triage this?

**Comment by rafiss** [comment]
Hi all, thanks for the patience with this. I have created a PR (#40) that should address this issue. We will need to find appropriate reviewers, so in the meantime, let us know if the workaround proposed by @dustinsmith1024 works for you.

**Comment by rafiss** [comment]
We have released v1.1.0 of sequelize-cockroachdb: https://www.npmjs.com/package/sequelize-cockroachdb

Please try it out and let us know how it goes!

**Comment by Sy1v4in** [comment]
Hello 👋

I think I have the same issue with the `Model.upsert` which seems to use the `CREATE OR REPLACE FUNCTION`.
